Add bulk addition to string list component

s0nea opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #9707

Add bulk addition option to string list component.

### Screenshot/Video
<!-- Attach screenshot or video of the changes and eventual comparison if you find it necessary -->
[rd.webm](https://github.com/rancher/dashboard/assets/8761082/5e341626-c84f-4b1e-907e-c6f3f779826e)

torchiaf commented on pkg/rancher-components/src/components/StringList/StringList.vue:
```diff
@@ -360,6 +356,47 @@ export default Vue.extend({
+     * @param index The list index of the item to be updated (optional).
+     * @returns Updated items list.
+     */
+    addValueToItems(items: string[], value: string, index?: number): string[] {
```
I think the `items` parameter may be decoupled from the return value here:
```
const newItems = [...items];
...
return newItems;
```
and change the input wherever `addValueToItems` is called: `addValueToItems([...this.items]) -> addValueToItems(this.items)`

torchiaf commented on pkg/rancher-components/src/components/StringList/StringList.vue:
```diff
@@ -343,12 +343,8 @@ export default Vue.extend({
-          items[index] = value;
-        }
+        const index = findStringIndex([...this.items], item, false);
+        const items = index !== -1 ? this.addValueToItems([...this.items], value, index) : [...this.items];
```
see comment on `addValueToItems` below

torchiaf commented on pkg/rancher-components/src/components/StringList/StringList.test.ts:
```diff
@@ -398,6 +398,241 @@ describe('stringList.vue', () => {
+        expect(JSON.stringify(itemsResult)).toBe(JSON.stringify(result));
+      });
+
+      it('should show warning if one of the values is a duplicate', async() => {
```
Could you please add a new `it()` block to test the case of duplicate items inside bulk items?
example:
```
const newValue = 'test;test';
```

torchiaf reviewed (changes requested):
The bulkable functionality works great, just a couple of comment about possible test/code imporvements

s0nea commented:
> The bulkable functionality works great, just a couple of comments about possible test/code improvements.

@torchiaf thank you for your review! I think I addressed all of your comments. Would you mind taking another look at it?

torchiaf reviewed (approved):
LGTM

rak-phillip reviewed (commented):
@s0nea can we bump the rancher components version so that we will be ready to publish a release once this merges?

s0nea commented:
> @s0nea can we bump the rancher components version so that we will be ready to publish a release once this merges?

@rak-phillip yes, thank you for mentioning it! I pushed an update.
